File: 1. Creational Design Patterns/Factory Method/C#/FactoryMethodTemplate.cs
// Factory Method Pattern – C# Template
using System;
namespace FactoryMethodTemplate
{
    interface IProduct { }
    abstract class Creator { public abstract IProduct FactoryMethod(); }
    class ConcreteCreator : Creator { public override IProduct FactoryMethod() => new ConcreteProduct(); }
    class ConcreteProduct : IProduct {}
    class Program { static void Main() { var creator = new ConcreteCreator(); var product = creator.FactoryMethod(); } }
}

File: 1. Creational Design Patterns/Factory Method/Kotlin/FactoryMethodTemplate.kt
// Factory Method – Kotlin Template
interface Product
abstract class Creator { abstract fun factoryMethod(): Product }
class ConcreteProduct: Product
class ConcreteCreator: Creator() { override fun factoryMethod() = ConcreteProduct() }
fun main(){ val creator: Creator = ConcreteCreator(); creator.factoryMethod() }
